### ISSUE TITLE
Move installation, usage, and FAQ into Sopel's documentation

### ISSUE THREAD

**Issue by Exirel** [sopel-irc/sopel#2497]
### Requested Feature

I'd like, for Sopel 8, to move part of the website's documentation into the project's documentation. The goal of this issue is to discuss which sections/parts we want exactly. I'd like to have a curated list of frequent issues and errors our users encountered in the past, things that we always have to answer to.

For instance, I'd like to improve the [running the bot](https://sopel.chat/docs/run.html) chapter with the following sections and information:

* how to install the bot (to replace [/usage/installing](https://sopel.chat/usage/installing/))
* Sopel's list of feature (including a description of the [per-channel feature](https://sopel.chat/usage/per-channel-configuration/))
* how to manage plugins, where are they installed, how to install more, etc.
* an FAQ with things like "my command doesn't run" and advice on how to troubleshoot some of the most common problems

I also think we could rework a bit our contribution guide, for example to add the appendix about the [core plugin](https://sopel.chat/appendix/core-plugin-guidelines/).

Maybe even... have a Design Document that outline the core tenet of Sopel's software architecture? Well... maybe not right now, but that would be lovely to consider!

As for the website, I think we could use it more to promote Sopel's feature, promote "official" plugins, share some links... I believe it still has a place and could be useful. It could also be used to host a sopel-help generated html page. That's something for another issue tho.

### Problems Solved

Today, there are two locations for our documentation, and that's one too many in my opinion. So moving everything (or at least the most relevant things) into one place would solve that issue. It would streamline the workflow of upgrading the doc at the same time as the code.

### Alternatives

Keep the website and upgrade it as we go, which we don't really do as it's separate from the code and all, and we have to keep the doc for the current version released, it means having open PR and stuff. So maybe we should change our workflow a bit to make it easier. But that's only the alternative here!

### Notes

I was supposed to make a PR for 2497 but I think it's better to talk about that before really diving into that. I'd love some feedback and ideas to write this part of the documentation.

**Comment by dgw** [comment]
mrw @Exirel adds something new to the 8.0.0 milestone:

https://github.com/sopel-irc/sopel/assets/164140/39a0d31f-d226-4cc6-b6e4-6bb8aa22d583

I'm not super attached to the current site, other than it being nostalgic (the basic structure is the same as what existed when I took over from the previous maintenance team). Automatic deployment of just the Sphinx docs is very much easier, as proven by our use of https://unstable.docs.sopel.chat/ (with deploy previews for every PR, and the main domain updated every time something's merged). I don't think it'd be too hard to shift sopel.chat itself over to auto-deploy only on release—maybe using another GHA workflow with https://github.com/netlify/actions.

> promote "official" plugins

Have a poke around https://github.com/sopel-irc/plugin-db (for now, the `prototype` branch) to see some past work regarding a proper plugin list. Fair warning: I last touched it almost 3 years ago, before I learned a lot more about static site generator tools. Regardless of what's under the hood and what still builds (or not), there is still a preview of the WIP version at https://prototype.plugins.sopel.chat/.

**Comment by Exirel** [comment]
Working on a first draft for the install guide.

![image](https://github.com/sopel-irc/sopel/assets/3259552/cec844c8-a3e9-4b9c-bd8a-d09c7662ff1f)

My approach is to do an Install Guide that is **not perfect**, i.e. it will contain the bare minimum to install Sopel, and it will be a way to let other people contribute to this install/usage/FAQ project.

Also, I believe we can totally live with a strip down version of what is currently available on the website, and increment it with Sopel 8.0.1 or 8.0.2 and other patch version.

**Comment by Exirel** [comment]
I did a thing...

![image](https://github.com/sopel-irc/sopel/assets/3259552/c645f9ba-8d99-495a-98fb-485c05fb5d5c)

I still need to detail the "managing plugins" section, so I could draft a PR right now. I don't think we need more at the moment, as everything else can be either fix in 8.0.1+ or updated in 8.1+.

**Comment by Exirel** [comment]
Note: with #2533 we have a basis for documentation, and we could probably consider it good enough for Sopel 8.0, and put this issue in Sopel 8.0.x or 8.1.x.

**Comment by Exirel** [comment]
Yup, I can confirm that #2533 is good enough for 8.0.0, and this issue could be pushed to 8.1.x for more!
